File: client/src/types/JsonModels/Analytics/AnalyticsOverviewModel.ts
import type { JsonModels } from "../../backend";

export interface AnalyticsOverviewModel
{
	workoutCount: number;
	totalVolumeKg: number;
	totalSets: number;
	totalReps: number;
	volumeTrend: JsonModels.Analytics.VolumeTrendPointModel[];
	frequentExercises: JsonModels.Analytics.FrequentExerciseSummaryModel[];
	muscleGroupVolumes: JsonModels.Analytics.MuscleGroupVolumeModel[];
	personalRecords: JsonModels.Analytics.PersonalRecordSummaryModel[];
}

File: server/FitMate.Core/JsonModels/Analytics/AnalyticsOverviewModel.cs
namespace FitMate.Core.JsonModels.Analytics;

public class AnalyticsOverviewModel
{
    public int WorkoutCount { get; set; }
    public decimal TotalVolumeKg { get; set; }
    public int TotalSets { get; set; }
    public int TotalReps { get; set; }
    public List<VolumeTrendPointModel> VolumeTrend { get; set; } = [];
    public List<MuscleGroupVolumeModel> MuscleGroupVolumes { get; set; } = [];
    public List<PersonalRecordSummaryModel> PersonalRecords { get; set; } = [];
}
